homeassistant/components/eq3btsmart/__init__.py

```diff
@@ -0,0 +1,138 @@
+"""Support for EQ3 devices."""
+
+import asyncio
+import logging
+
+from eq3btsmart import Thermostat
+from eq3btsmart.exceptions import Eq3Exception
+from eq3btsmart.thermostat_config import ThermostatConfig
+
+from homeassistant.components import bluetooth
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import CONF_MAC, CONF_NAME, Platform
+from homeassistant.core import HomeAssistant
+from homeassistant.exceptions import ConfigEntryNotReady
+from homeassistant.helpers.dispatcher import async_dispatcher_send
+
+from .const import DOMAIN, SIGNAL_THERMOSTAT_CONNECTED, SIGNAL_THERMOSTAT_DISCONNECTED
+from .models import Eq3Config, Eq3ConfigEntry
+
+PLATFORMS = [
+    Platform.CLIMATE,
+]
+
+_LOGGER = logging.getLogger(__name__)
+
+
+async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
+    """Handle config entry setup."""
+
+    mac_address: str = entry.data[CONF_MAC]
+    name: str = entry.data[CONF_NAME]
+
+    eq3_config = Eq3Config(
+        mac_address=mac_address,
+        name=name,
+    )
+
+    device = bluetooth.async_ble_device_from_address(
+        hass, mac_address, connectable=True
+    )
+
+    if device is None:
+        raise ConfigEntryNotReady(f"[{eq3_config.name}] Device could not be found")
+
+    thermostat = Thermostat(
+        thermostat_config=ThermostatConfig(
+            mac_address=mac_address,
+            name=name,
+        ),
+        ble_device=device,
+    )
+
+    eq3_config_entry = Eq3ConfigEntry(eq3_config=eq3_config, thermostat=thermostat)
```

Comment:
```suggestion
    eq3_config_entry = Eq3ConfigEntry(eq3_config=eq3_config, thermostat=thermostat)
```

Let's call this `Eq3ConfigEntryData` so its clear what it is